### ISSUE TITLE
Flexget v2.0.6 and urlrewriting

### ISSUE THREAD

**Issue by Lukeid** [Flexget/Flexget#1104]
Hello,

Since I updated to v2.0.x I'm having problems with url rewriting. I can manage to rewrite successful a url but after the rewriting Flexget keep using the original url. I tested it with download and transmission plugins.

Transmission throws an error of not valid torrent file:

`2016-04-28 12:59 ERROR    transmission  new-pct1        TransmissionError: Query failed with result "invalid or corrupt torrent file"
`

And Download plugin download the HTML page referenced in the RSS:
```
2016-04-28 18:47 DEBUG    urlrewriter   new-pct1        Url rewriting http://newpct1.com/serie/11-22-63/capitulo-104/
2016-04-28 18:47 DEBUG    utils.requests new-pct1        Fetching http://newpct1.com/descarga-torrent/serie/11-22-63/capitulo-104/
2016-04-28 18:47 TRACE    entry         new-pct1        ENTRY SET: url = u'http://www.newpct1.com/descargar-torrent/1461828655_11-22-63---temporada-1--hdtv/'
2016-04-28 18:47 INFO     urlrewriter   new-pct1        Entry '11 22 63 -  [HDTV][1x04][Español Castellano]' URL rewritten to http://www.newpct1.com/descargar-torrent/1461828655_11-22-63---temporada-1--hdtv/ (with newpct1)
2016-04-28 18:47 TRACE    urlrewriter   new-pct1        checking urlrewriter eztv
...
2016-04-28 18:47 TRACE    urlrewriter   new-pct1        checking urlrewriter bakabt
2016-04-28 18:47 INFO     download      new-pct1        Downloading: 11 22 63 -  [HDTV][1x04][Español Castellano]
2016-04-28 18:47 DEBUG    download      new-pct1        Downloading url 'http://newpct1.com/serie/11-22-63/capitulo-104/'
```
The url in the RSS is 'http://newpct1.com/serie/11-22-63/capitulo-104/' and it seems it is correctly rewrited to 'http://www.newpct1.com/descargar-torrent/1461828655_11-22-63---temporada-1--hdtv/' but after that it download the original url 'http://newpct1.com/serie/11-22-63/capitulo-104/'.

I've verified that the file downloaded corresponds to original url and that if I do a wget of the rewritten url it download the torrent file correctly.

Any ideas?

Thank you very much

**Comment by Lukeid** [comment]
I've investigate a little more. The issue doesn't seem related with v2.0.x. It is reproducible in 1.2.521.

I don't know exactly which is the root cause but I could trace it to this piece of code in download.py:

```
    def get_temp_file(self, task, entry, require_path=False, handle_magnets=False, fail_html=True,
                      tmp_path=tempfile.gettempdir()):
        if entry.get('urls'):
            urls = entry.get('urls')
        else:
            urls = [entry['url']]
        errors = []
```

It sets urls as "entry.get('urls')" which means original url 'http://newpct1.com/serie/11-22-63/capitulo-104/'. But I've check that correct one would be  "urls = [entry['url']" which cointains the rewritten url.

Does anybody know why is this happening?

Thanks in advance.

**Comment by crawln45** [comment]
Probably be better if you also posted your config. Seeing as everytime I parse that url. That's not what's returned for me.

**Comment by Lukeid** [comment]
Yes, sorry. I use a custom urlrewritter:
[newpct1.py.txt](https://github.com/Flexget/Flexget/files/243101/newpct1.py.txt)

And my config is like:
[config.ylm.txt](https://github.com/Flexget/Flexget/files/243113/config.ylm.txt)

Where series.yml is a file with tv shows list.

Hope this helps. :)

**Comment by crawln45** [comment]
Well it helps. but I return the correct values.
```
2016-04-29 14:55 DEBUG    urlrewriter   test            Url rewriting http://www.newpct1.com/serie/11-22-63/capitulo-104/
2016-04-29 14:56 DEBUG    utils.requests test            Fetching http://www.newpct1.com/descarga-torrent/serie/11-22-63/capitulo-104/
2016-04-29 14:56 INFO     urlrewriter   test            Entry '11.22.63.S01E04' URL rewritten to http://www.newpct1.com/descargar-torrent/1461828655_11-22-63---temporada-1--hdtv (with newpct1)
2016-04-29 14:57 INFO     download      test            Would download: 11.22.63.S01E04
2016-04-29 14:57 DEBUG    download      test            Successfully retrieved 11.22.63.S01E04 from http://www.newpct1.com/descargar-torrent/1461828655_11-22-63---temporada-1--hdtv
```

**Comment by crawln45** [comment]
Well it's not ideal until I can actually figure out a better way of updating 'urls'. For now you can probably just do this. https://bpaste.net/show/1d256f79643b for newpct1.py

**Comment by crawln45** [comment]
@gazpachoking any thoughts on this. seems we need to update both entry['url'] and entry['urls'] at the same time

**Comment by crawln45** [comment]
There got a fix in place. Will merge once it's done. Seems that when you use "link: alternate" for urls in rss feeds it creates a list of urls thinking you might want multiple urls to download from. But we didn't update the list of urls when we rewrote a url.

**Comment by crawln45** [comment]
Closed #1112

**Comment by Lukeid** [comment]
I confirm that version 2.0.8 solved the issue.

Thank you.
